addons/website_slides/static/src/js/slides_course_fullscreen_player.js

```diff
@@ -15,309 +15,431 @@ odoo.define('website_slides.fullscreen', function (require) {
                 self._goToNextSlide();
             }
         },
         /**
+         * Instanciate the quiz widget that will handle all the quiz functionalities
+         *
          * @private
-         * Creates slides objects from every slide-list-cells attributes
          */
-        _getSlides: function(){
+        _renderQuiz: function (){
             var self = this;
-            var slides = $('.o_wslides_fullscreen_sidebar_slide_tab');
-            for(var i = 0; i < slides.length;i++){
-                var slide = $(slides[i]);
-                self.slides.push({
-                    id: parseInt(slide.attr('slide_id'), 10),
-                    name: slide.attr('slide_name'),
-                    embed_code: slide.attr('slide_embed_code'),
-                    slide_type: slide.attr('slide_type'),
-                    done: slide.attr('done'),
-                    hasQuiz: slide.attr('quiz'),
-                    slug: slide.attr('slide_slug'),
-                    htmlContent: undefined
-                });
-                this._getActiveSlide();
-            }
+            var Quiz = new QuizWidget(this, self.slide, self.nextSlide);
+            Quiz.appendTo('.o_wslides_fs_player');
+            $('.next-slide').click(function (){
+                self._goToNextSlide();
+            });
+            $('.back-to-video').click(function (){
+                self.goToQuiz = false;
+                self._renderSlideContent();
+            });
         },
         /**
+         * Renders the player accordingly to the current slide
+         *
          * @private
-         * @param {object} slide
-         * Fetch the quiz for a particular slide
          */
-        _fetchQuiz: function(){
-            var self = this;
-            self._rpc({
-                route:"/slide/quiz/get",
-                params: {
-                    'slide_id': self.slide.id
-                }
-            }).then(function(data){
-                if(data){
-                    self.slide.quiz = data;
-                    self._renderPlayer();
-                }
-            })
-        },
-        _fetchHtmlContent: function(){
+        _renderSlideContent: function (){
             var self = this;
-            self._rpc({
-                route:"/slide/html_content/get",
-                params: {
-                    'slide_id': self.slide.id
-                }
-            }).then(function(data){
-                if (data.html_content) {
-                    self.slide.htmlContent = data.html_content;
-                    self._renderPlayer();
-                }
-            })
+            var embed_url = $(this.slide.embed_code).attr('src');
+            if (self.slide.slide_type === "video"){
+                embed_url = "https://" + embed_url + "&rel=0&autoplay=1&enablejsapi=1&origin=" + window.location.origin;
+            }
+            $('.o_wslides_fs_player').html(QWeb.render('website.slides.fullscreen', {
```

Comment:
When rendering sub part of widget, try to only give `{'widget': this}` as render value. So in the template you can access `widget.whatever_you_want` ;)